### PR TITLE
Fix DistributionSet type should not be updatable

### PR DESCRIPTION
**Pull request by blomark**
For an existing DistributionSet the type attribute should not be updatable. The Combobox showing the type of a Distributionset is now readonly in the Deployment View and Distribution View.

Reviewers: @stefbehl

### CODE REVIEW

File: hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/builder/JpaDistributionSetCreate.java [hunk 1]
```diff
@@ -29,6 +31,9 @@
 public class JpaDistributionSetCreate extends AbstractDistributionSetUpdateCreate<DistributionSetCreate>
         implements DistributionSetCreate {
 
+    @ValidString
+    protected String type;
```
Comment (blomark) [inline hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/builder/JpaDistributionSetCreate.java]: use private

File: hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/distributions/dstable/DistributionSetTableHeader.java [hunk 2]
```diff
@@ -98,7 +98,7 @@ protected void searchBy(final String newSearchText) {
 
     @Override
     protected void addNewItem(final ClickEvent event) {
-        final Window newDistWindow = addUpdateWindowLayout.getWindow(null);
+        final Window newDistWindow = addUpdateWindowLayout.getWindowForCreation();
```
Comment (blomark) [inline hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/distributions/dstable/DistributionSetTableHeader.java]: getWindowForCreate

File: hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/management/dstable/DistributionAddUpdateWindowLayout.java [hunk 3]
```diff
@@ -277,53 +277,64 @@ private DistributionSetType getDefaultDistributionSetType() {
...
-    public CommonDialogWindow getWindow(final Long editDistId) {
+    public CommonDialogWindow getWindowForUpdate(final Long editDistId) {
         resetComponents();
         populateDistSetTypeNameCombo();
         populateValuesOfDistribution(editDistId);
         return new WindowBuilder(SPUIDefinitions.CREATE_UPDATE_WINDOW)
                 .caption(i18n.getMessage(UIComponentIdProvider.DIST_ADD_CAPTION)).content(this).layout(formLayout)
-                .i18n(i18n).saveDialogCloseListener(new SaveOnCloseDialogListener()).buildCommonDialogWindow();
+                .i18n(i18n).saveDialogCloseListener(new UpdateOnCloseDialogListener(editDistId))
+                .buildCommonDialogWindow();
```
Comment (blomark) [inline hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/management/dstable/DistributionAddUpdateWindowLayout.java]: private getWindow method

File: hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/management/dstable/DistributionAddUpdateWindowLayout.java [hunk 4]
```diff
@@ -176,31 +174,33 @@ private void addNewDistribution() {
...
+        if (existingDs.isPresent() && !existingDs.get().getId().equals(editDistId)) {
+            distNameTextField.addStyleName(SPUIStyleDefinitions.SP_TEXTFIELD_LAYOUT_ERROR_HIGHTLIGHT);
+            distVersionTextField.addStyleName(SPUIStyleDefinitions.SP_TEXTFIELD_LAYOUT_ERROR_HIGHTLIGHT);
+            notificationMessage.displayValidationError(i18n.getMessage("message.duplicate.dist",
+                    new Object[] { existingDs.get().getName(), existingDs.get().getVersion() }));
 
+            return true;
         }
 
+        return false;
```
Comment (stefbehl) [inline hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/management/dstable/DistributionAddUpdateWindowLayout.java]: You could avoid the get() calls on the Optional like this:

        return existingDs.filter(ds -> ds.getId().equals(editDistId)).map(ds -> {
            distNameTextField.addStyleName(SPUIStyleDefinitions.SP_TEXTFIELD_LAYOUT_ERROR_HIGHTLIGHT);
          distVersionTextField.addStyleName(SPUIStyleDefinitions.SP_TEXTFIELD_LAYOUT_ERROR_HIGHTLIGHT);
            notificationMessage.displayValidationError(i18n.getMessage("message.duplicate.dist",
                    new Object[] { existingDs.get().getName(), existingDs.get().getVersion() }));
            return true;
        }).orElse(false);

**Pull request merged**